File: Lessons/Lesson 1 Time Complexity/TapeEquilibrium/TapeEq.cpp
//Codility Lesson 1, Program 1 TapeEq

#include<iostream>
#include<limits.h>
#include<cmath>
#include<vector>
using namespace std;

int solution(vector<int> &A) {
	int lsum = A[0], rsum = 0, min;
	int size = A.size();
    for(int i=1; i<size; ++i){
    	rsum += A[i];
    	cout<<A[i]<<endl;
    }
    min= abs(lsum-rsum);
    if(A.size()==2){
        return min;
    }
    for(int i=1; i<size-1; ++i){
    	lsum += A[i];
		rsum -= A[i];
		int newmin = lsum - rsum;
		if(min > (abs(newmin))){
			min = (abs(newmin));
			cout<<"newMin: "<<min<<" i: "<<i<<" lsum: "<<lsum<<" rsum: "<<rsum<<endl;
		}
    }
    return min;
}

File: Lessons/Lesson 1 Time Complexity/TapeEquilibrium/TapeEq.pl
# you can use print for debugging purposes, e.g.
# print "this is a debug message\n";

sub solution {
    my (@A)=@_;
    # write your code in Perl 5.14
    my $left = $A[0];
    my $right = 0;
    for (my $i = 1; $i < scalar @A; $i++){
        $right += $A[$i];
    }
    my $min = abs($right - $left);
    if((scalar @A) == 2){
        return $min;
    }
    for (my $i = 1; $i < (scalar @A)-1; $i++){
        $left += $A[$i];
        $right -= $A[$i];
        my $newmin = $left - $right;
        if($min > abs($newmin)){
            $min = abs($newmin);
        }
    }
    return $min;
}